sdk/storage/Azure.Storage.Common.DataMovement/src/FilesystemScanner.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Azure.Storage.Common.DataMovement
+{
+    /// <summary>
+    /// FilesystemScanner class.
+    /// </summary>
+    public static class FilesystemScanner
+    {
+        /// <summary>
+        /// Enumerates all files pointed to by the provided path, including those in subdirectories (if path is a directory).
+        /// </summary>
+        /// <param name="path">Filesystem location.</param>
+        /// <returns>Enumerable list of absolute paths containing all relevant files the user has permission to access.</returns>
+        public static IEnumerable<string> ScanLocation(string path)
```

Comment:
https://objcsharp.wordpress.com/2013/07/08/why-static-code-is-bad/

Reply:
#21715 I refactored the scanner to make it non-static and shift a couple things around, let me know if there's any issues with this implementation.